mlflow/utils/_capture_modules.py

```diff
@@ -111,14 +111,31 @@ def main():
     # Mirror `sys.path` of the parent process
     sys.path = json.loads(args.sys_path)
 
-    if flavor == mlflow.spark.FLAVOR_NAME and is_in_databricks_runtime():
+    if flavor == mlflow.spark.FLAVOR_NAME:
         # Clear 'PYSPARK_GATEWAY_PORT' and 'PYSPARK_GATEWAY_SECRET' to enforce launching a new JVM
         # gateway before calling `mlflow.spark._load_pyfunc` that creates a new spark session
         # if it doesn't exist.
         os.environ.pop("PYSPARK_GATEWAY_PORT", None)
         os.environ.pop("PYSPARK_GATEWAY_SECRET", None)
+
+    if flavor == mlflow.spark.FLAVOR_NAME and is_in_databricks_runtime():
         os.environ["SPARK_DIST_CLASSPATH"] = "/databricks/jars/*"
 
+    if flavor == mlflow.spark.FLAVOR_NAME and not is_in_databricks_runtime():
+        # Create a local spark environment within the subprocess if using OSS Spark
+        from pyspark.sql import SparkSession
+
+        SparkSession.builder.config("spark.python.worker.reuse", "true").config(
+            "spark.databricks.io.cache.enabled", "false"
+        ).config("spark.executor.allowSparkContext", "true").config(
+            "spark.driver.bindAddress", "127.0.0.1"
+        ).config(
+            "spark.driver.extraJavaOptions",
+            "-Dlog4j.configuration=file:/usr/local/spark/conf/log4j.properties",
+        ).master(
```

Comment:
Nit: Can we format this so that each `config()` call occurs on separate line?

Reply:
definitely!